tests/pkg/asserters/formation_assignments_async.go

```diff
@@ -0,0 +1,65 @@
+package asserters
+
+import (
+	"context"
+	"github.com/kyma-incubator/compass/components/director/pkg/str"
+	context_keys "github.com/kyma-incubator/compass/tests/pkg/context-keys"
+	"github.com/kyma-incubator/compass/tests/pkg/fixtures"
+	"github.com/machinebox/graphql"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+)
+
+type FormationAssignmentsAsyncAsserter struct {
+	FormationAssignmentsAsserter
+	delay int64
+}
+
+func NewFormationAssignmentAsyncAsserter(expectations map[string]map[string]fixtures.AssignmentState, expectedAssignmentsCount int, certSecuredGraphQLClient *graphql.Client, tenantID string, delay int64) *FormationAssignmentsAsyncAsserter {
+	f := FormationAssignmentsAsyncAsserter{
+		FormationAssignmentsAsserter: FormationAssignmentsAsserter{
+			expectations:             expectations,
+			expectedAssignmentsCount: expectedAssignmentsCount,
+			certSecuredGraphQLClient: certSecuredGraphQLClient,
+			tenantID:                 tenantID,
+		},
+		delay: delay,
+	}
+	return &f
+}
+
+func (a *FormationAssignmentsAsyncAsserter) AssertExpectations(t *testing.T, ctx context.Context) {
+	formationID := ctx.Value(context_keys.FormationIDKey).(string)
+	a.assertFormationAssignmentsAsynchronously(t, ctx, a.certSecuredGraphQLClient, a.tenantID, formationID, a.expectedAssignmentsCount, a.expectations)
+}
+
+func (a *FormationAssignmentsAsyncAsserter) assertFormationAssignmentsAsynchronously(t *testing.T, ctx context.Context, certSecuredGraphQLClient *graphql.Client, tenantID, formationID string, expectedAssignmentsCount int, expectedAssignments map[string]map[string]fixtures.AssignmentState) {
```

Comment:
The eventually method has several drawbacks - In case of error the test will stuck for 60 seconds instead of 8-16. If the current assert method is adapted to use eventually it will become too crowded with if-sattements also there will be a lot of logs from assertion attempts from a previous tick. Also in case of failure you will have to go and see the logs above instead of just look at the error message. 

The fixed delays can be shortened to 100-200ms after the PR is merged and the blocked PR with the default responder is validated. We can check if further micro optimisations areneeded after that.

Reply:
The eventually assert can be refactored to produce fewer logs and used when the timeouts are lowered as we have to keep the tick period to a minimum to save time and not produce lots of logs at the same time.